dev/io.openliberty.jakartaee9.internal_fat/fat/src/io/openliberty/jakartaee9/internal/tests/EE9FeatureCompatibilityTest.java

```diff
@@ -0,0 +1,311 @@
+                String line = scanner.nextLine();
+                if (line.startsWith("IBM-ShortName:")) {
+                    shortName = line.substring("IBM-ShortName:".length()).trim();
+                } else if (line.contains("IBM-Test-Feature:") && line.contains("true")) {
+                    Log.info(c, "parseShortName", "Skipping test feature: " + feature.getName());
+                    return;
+                } else if (line.startsWith("Subsystem-SymbolicName:") && !line.contains("visibility:=public")) {
+                    Log.info(c, "parseShortName", "Skipping non-public feature: " + feature.getName());
+                    return;
+                } else if (line.startsWith("IBM-ProductID") && !line.contains("io.openliberty")) {
+                    Log.info(c, "parseShortName", "Skipping non Open Liberty feature: " + feature.getName());
+                    return;
+                }
+            }
+            features.add(shortName);
+        } finally {
+            scanner.close();
+        }
+    }
+
+    /**
+     * This test is marked as FULL FAT because it can take 10 to 30 minutes to run depending on
+     * the system. This puts it past the 5 minute expected limit for lite tests. The jsonp-2.0 test
+     * is a quick test to show that the basics work with feature resolution.
+     *
+     * @throws Exception
+     */
+    @Test
+    @Mode(TestMode.FULL)
+    public void testJakarta91ConvenienceFeature() throws Exception {
+        Map<String, String> specialEE9Conflicts = new HashMap<>();
+        // faces and facesContainer conflict with each other
+        specialEE9Conflicts.put("facesContainer-3.0", "io.openliberty.facesProvider");
+        // the jakartaee-9.1 convenience feature conflicts with itself and with 9.0
+        specialEE9Conflicts.put("jakartaee-9.0", "io.openliberty.jakartaee");
+        specialEE9Conflicts.put("jakartaee-9.1", "io.openliberty.jakartaee");
+        // the convenience feature depends on jdbc-4.2 and tolerates 4.3
+        specialEE9Conflicts.put("jdbc-4.0", "com.ibm.websphere.appserver.jdbc");
+        specialEE9Conflicts.put("jdbc-4.1", "com.ibm.websphere.appserver.jdbc");
+        // the webProfile-9.1 convenience feature conflicts with the 9.0 one
+        specialEE9Conflicts.put("webProfile-9.0", "io.openliberty.webProfile");
+
+        testCompatibility("jakartaee-9.1", specialEE9Conflicts);
+    }
+
+    @Test
+    public void testJsonP20Feature() throws Exception {
+        // jsonp-2.0 will conflict with itself
+        testCompatibility("jsonp-2.0", Collections.singletonMap("jsonp-2.0", "io.openliberty.jsonp"));
+    }
+
+    private void testCompatibility(String featureName, Map<String, String> specialConflicts) throws Exception {
+        final List<String> errors = new CopyOnWriteArrayList<>();
+        Result jakartaee91 = resolver.resolveFeatures(repository, Collections.<ProvisioningFeatureDefinition> emptySet(), Collections.singletonList("jakartaee-9.1"),
```

Comment:
Seems this resolution is only here to get the set of kernel features.  Perhaps resolve something much more simple (e.g. osgiConsole-1.0).  But I'm not sure I understand how this code gets the kernel features though because nothing explicitly includes the kernel features that I know of so how would a resolve pull it in?

Reply:
It gives me an empty list.  As such I am just removing this code.